### PR TITLE
[ADD] partner_address_format: New module welcome.

### PR DESCRIPTION
**Pull request by nhomar**
review the README.rst to understand the rationale of this new module.

### CODE REVIEW

File: partner_address_format/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    """,
+    'author': "Vauxoo",
+    'website': "http://www.vauxoo.com",
+    'category': 'Base',
+    'version': '0.1',
+    'depends': [
+        'portal',
+    ],
+    'data': [
+        # 'security/ir.model.access.csv',
```
Comment (moylop260) [inline partner_address_format/__openerp__.py]: Is necessary this comment?

File: partner_address_format/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+{
+    'name': "Partner Html Address Format",
+    'summary': """Customize the way you show the address in html to have a
+    proper global way in odoo to do that.""",
+    'description': """
+        Long description of module's purpose
```
Comment (moylop260) [inline partner_address_format/__openerp__.py]: Do you need `description` key and `Readme` file?

**Comment by moylop260** [comment]
👍 
Greats module!

just small comments
Thanks

File: partner_address_format/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+{
+    'name': "Partner Html Address Format",
+    'summary': """Customize the way you show the address in html to have a
+    proper global way in Odoo to do that.""",
+    'author': "Vauxoo",
+    'website': "http://www.vauxoo.com",
+    'category': 'Base',
+    'version': '0.1',
```
Comment (moylop260) [inline partner_address_format/__openerp__.py]: Use 8.0.X.Y.Z

File: partner_address_format/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        'portal',
+    ],
+    'data': [
+        'templates.xml',
+        'view.xml',
+        'data.xml',
+    ],
+    'demo': [
+        'demo.xml',
+    ],
```
Comment (moylop260) [inline partner_address_format/__openerp__.py:32]: Add `installable` and `license` keys

File: partner_address_format/tests/test_get_address.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        self.contact_use_parent = self.env.ref('base.res_partner_2')
+        # Agrolait,
+        # Thomas Passot To test a contact with a company.
+        self.company_no_parent = self.env.ref('base.res_partner_address_3')
+        # Demo portal User: To test a contact without a company as the portal
+        # users ones.
+        self.contact_no_company = self.env.ref('portal.partner_demo_portal')
+        self.mexico = self.env.ref('base.mx')
+        self.main_company = self.env.ref('base.main_company')
+        self.view_on_company = self.env.ref('partner_address_format.company_format')
```
Comment (moylop260) [inline partner_address_format/tests/test_get_address.py]: FYI flake8 checks:

``` bash
partner_address_format/models.py:83:1: W391 blank line at end of file
partner_address_format/tests/test_get_address.py:22:80: E501 line too long (84 > 79 characters)
partner_address_format/tests/test_get_address.py:23:80: E501 line too long (86 > 79 characters)
partner_address_format/tests/test_get_address.py:79:70: E231 missing whitespace after ','
```

File: partner_address_format/models.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, fields, api
+from openerp.addons.base.ir.ir_qweb import QWebException
+
+
+class ResCompany(models.Model):
```
Comment (moylop260) [inline partner_address_format/models.py]: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming
Use `models/<main_model>.py`

File: partner_address_format/models.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                                        help='A normal view with the format to'
+                                             'be used to format the address '
+                                             'for this company.')
+
+
+class ResCountry(models.Model):
+    _inherit = 'res.country'
+
+    address_format_id = fields.Many2one('ir.ui.view',
+                                        help='A normal view with the format to'
```
Comment (moylop260) [inline partner_address_format/models.py]: `s/A normal view with the format to format/A normal view with the template to format/g`

**Comment by nhomar** [comment]
@moylop260 Ready for merge in order to use it into the apps modules.

No report yet is using this new api, I expect start to propose reports using it once it is merged (any comment is welcome).

File: partner_address_format/models/res_country.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# @author Nhomar Hernandez <nhomar@vauxoo.com>
+from openerp import models, fields
+
+
+class ResCountry(models.Model):
+    _inherit = 'res.country'
+
+    address_format_id = fields.Many2one('ir.ui.view',
+                                        help='A normal view with the format to'
```
Comment (moylop260) [inline partner_address_format/models/res_country.py]: Remove this lines

File: partner_address_format/models/res_company.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# @author Nhomar Hernandez <nhomar@vauxoo.com>
+from openerp import models, fields
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    address_format_id = fields.Many2one('ir.ui.view',
+                                        help='A normal view with the format to'
```
Comment (moylop260) [inline partner_address_format/models/res_company.py]: Remove this lines

File: partner_address_format/data/data.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
```
Comment (moylop260) [inline partner_address_format/data/data.xml]: Could we rename the file to model_data.xml?

File: partner_address_format/demo/demo.xml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
```
Comment (moylop260) [inline partner_address_format/demo/demo.xml]: Could we rename the file to model_demo.xml?

File: partner_address_format/__openerp__.py [hunk 12]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
```
Comment (moylop260) [inline partner_address_format/__openerp__.py]: Add license comment

**Comment by moylop260** [comment]
I'll work my feedback in this branch...
Do you know if the red is unrelated?

File: partner_address_format/__openerp__.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+{
+    'name': "Partner Html Address Format",
+    'summary': """Customize the way you show the address in html to have a
+    proper global way in Odoo to do that.""",
+    'author': "Vauxoo",
+    'website': "http://www.vauxoo.com",
+    'category': 'Base',
+    'version': '8.0.0.1.0',
+    'depends': [
+        'portal',
```
Comment (moylop260) [inline partner_address_format/__openerp__.py:21]: I found this dependency used just for tests case...
Is this right?

File: partner_address_format/models/res_partner.py [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+        """
+        context = context is None and {} or context
+        view = self.pick_view(cr, uid, ids, context=context)
+        val = {'partner': self.browse(cr, uid, ids[0], context=context), }
+        try:
+            # Catch any error in declaration of the view, if the user make an
+            # error it will help in the debugging process.
+            view_obj = self.pool['ir.ui.view']
+            html = view_obj.render(cr, uid, view.id, val, context=context)
+        except QWebException, exc:
```
Comment (moylop260) [inline partner_address_format/models/res_partner.py]: Uses a deprecated syntax that is ambiguous with except (ExceptionOne, ExceptionTwo). as would be better than the comma.

File: partner_address_format/tests/test_get_address.py [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+        self.assertNotIn('QWebException', html,
+                         'Template: %s  .' % xml_id)
+        html = self.company_no_parent.with_context(
+            ctx).get_address_from_template()
+        self.assertNotIn('QWebException', html,
+                         'Template: %s  .' % xml_id)
+
+    def test_020_address_templates(self):
+        """Testing all templates on this module."""
+        [self.by_template_id(template) for template in self.templates]
```
Comment (moylop260) [inline partner_address_format/tests/test_get_address.py:124]: Should we add a assert?

**Comment by moylop260** [comment]
@nhomar 
Please, check my last commits

Last comments are:
1. I found `portal` dependency used just for tests case... Is this right?
2. The `test_020_address_templates` don't have `assert` and have a expression assigned to nothing.

What do you think?

File: partner_address_format/tests/test_get_address.py (hunk 15 above)
Comment (nhomar) [inline partner_address_format/tests/test_get_address.py:124]: Asserts are generic ones, this method is to trigger a test per template (asserts into by_template_id) do you think there is a more ellegant way to achieve this?

Comment (moylop260) [inline partner_address_format/tests/test_get_address.py:124]: I got it
Thank you

File: partner_address_format/models/res_company.py [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+# © 2016 Vauxoo (http://www.vauxoo.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# @author Nhomar Hernandez <nhomar@vauxoo.com>
+from openerp import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    address_format_id = fields.Many2one('ir.ui.view',
```
Comment (moylop260) [inline partner_address_format/models/res_company.py:11]: We could use all type of views or Should we add a `qweb` domain?

Comment (nhomar) [inline partner_address_format/models/res_company.py:11]: done

**Comment by moylop260** [comment]
LGTM

**Pull request merged**